plugins/README.adoc

```diff
@@ -7,3 +7,256 @@ NOTE: Work in Progress...
+This keeps the plugin code easier to read and maintain.
+500 lines is a general guideline that should serve most cases.
+====
+
+== Action plugins should be used when the code runs on the control node.
+[%collapsible]
+====
+Explanations::
+Use an action plugin for code that needs to run on the controller, where the host would be `localhost`.
+Rationale::
+Action plugins let you integrate local processing and local data with module functionality.
+Action plugins act in conjunction with modules to execute the actions required by playbook tasks.
+Using an action plugin will cause Ansible to skip creating a temporary tarball of execution code and runs the code directly on the controller host.
+
+Examples::
+See the `ansible.snmp.get` https://github.com/ansible-network/ansible.snmp/blob/main/plugins/action/get.py[action plugin] and associated https://github.com/ansible-network/ansible.snmp/blob/main/plugins/modules/get.py[module].
+
+====
+
+== Design decisions accounting for “jump hosts” should be avoided.
+[%collapsible]
+====
+Explanations::
+A plugin should be developed in accordance with Ansible best practices, without accounting for possible environment-specific variables.
+
+Rationale::
+The plugin author should prioritize development of the plugin's function.
+If an environment that the plugin could possibly run in (such as on a jump host) has some unique property this should be managed by the operator's Ansible configuration, separate from the plugin code.
+
+====
+
+== Plugins should be initially developed using the ansible plugin builder
+[%collapsible]
+====
+Explanations::
+The https://github.com/ansible-community/ansible.plugin_builder[ansible.plugin_builder] is a tool which helps developers scaffold new plugins.
+
+====
+
+== Pylint and type exceptions should have a specific exclusion type
+[%collapsible]
+====
+Explanations::
+Pylint and type exceptions should only be used when required due to bugs or 3rd party package requirements. All should have a specific exclusion type.
+
+Rationale::
+Linting and type exceptions should not be used except where the logic of the code absolutely requires it.
+
+Examples::
+[source,python]
+----
+            if not rule.get('group_desc', '').strip():
+                # retry describing the group once
+                try:
+                    auto_group = get_security_groups_with_backoff(client, Filters=ansible_dict_to_boto3_filter_list(filters)).get('SecurityGroups', [])[0]
+                except (is_boto3_error_code('InvalidGroup.NotFound'), IndexError):
+                    module.fail_json(msg="group %s will be automatically created by rule %s but "
+                                         "no description was provided" % (group_name, rule))
+                except ClientError as e:  # pylint: disable=duplicate-except
+                    module.fail_json_aws(e)
+----
+
+
+====
+
+== Plugins for applicance configuration should follow the Resource Module pattern (see below)
```

Comment:
This one seems very very specific, and I don't really understand it. As it seems also that you're just pointing at existing documentation, I would skip it. We (IMHO) don't need to do deep-linking to the official documentation.

Reply:
Removing this section